### DISCUSSION TITLE
Having trouble requesting different outputs from regtests

### DISCUSSION THREAD

**Discussion by japham0** [NOAA-EMC/WW3#1235] [Q&A]
I have only just gotten started using Wavewatch III, and I have already referred to the manual but am still confused and unclear. 
I have run the regtest (for example, ww3_tp2.2) by doing:

./ww3_grid > ww3_grid.out
./ww3_strt > ww3_strt.out
./ww3_shel > ww3_shel.out
./ww3_ounf > ww3_ounf.out

and I see that one of the outputs is HS. I want to modify the test to also return LM (mean wave length). I went to the ww3_shel.inp file and added LM after HS to the script where it shows the line:

N
DPT CUR WND HS 

to 

N
DPT CUR WND HS LM

but I have not been able to see the LM in the wc.200206.nc file which is generated after running the 4 scripts.

Is this the correct way to go about requesting different outputs from the table to be returned (pg 410 of the manual PDF, G.52), aka just going to the ww3_shel.inp script and adding the specified acronym? Or is there more that we need to modify? Do we also need to modify any other scripts to request another output? I have tried modifying the above line in multiple ways (deleting and adding different coded acronyms) and have not been able to change the output when I rerun the 4 scripts required for the regtest.

I would really appreciate any detailed response, as I am new to wavewatch and am trying to learn. I have been reading the manual but it is quite verbose and confusing, and I would greatly appreciate any simplified explanation just so I can get started off. Thank you in advance for your help.

**Comment by mickaelaccensi** [comment]
it is the correct way, you just missed to also add LM in ww3_ounf.inp for the post-processing.

Best

**Reply by japham0** [reply]
Hello, thank you for your comment. This is what I have in ww3_shel.inp (for the regtest ww3_tp2.1). I simply vi ww3_shel.inp and modify the line to have "HS LM" and then save the changes.
   F T
   F T
   F T
   F F
   F F
   F F
   F
   F
   F
$
   19680606 000000
   19680606 050000
$  19680608 000000
$
   1
$
   19680606 000000    360  19680608 000000
$
N
HS LM
$
   19680606 000000      0  19680608 000000
   19680606 000000      0  19680608 000000
   19680606 000000      0  19680608 000000
   19680606 000000      0  19680608 000000
   19680606 000000      0  19680608 000000

And this is what I have for ww3_ounf.inp, after simpy doing vi ww3_ounf.inp and similarly saving the changes "HS LM."

 -------------------------------------------------------------------- $
$ WAVEWATCH III Grid output post-processing                            $
$--------------------------------------------------------------------- $
$ First output time (yyyymmdd hhmmss), increment of output (s),
$ and number of output times.
$
  19680606 000000 1080. 17
$
$ Fields requested --------------------------------------------------- $
$
$ Output request flags identifying fields as in ww3_shel.inp. See that
$ file for a full documentation of field output options. Namelist type
$ selection is used here (for alternative F/T flags, see ww3_shel.inp).
$
 N
HS LM
$
$--------------------------------------------------------------------- $
$ netCDF version [3,4]
$        and variable type 4 [2 = SHORT, 3 = it depends , 4 = REAL]
$ swell partitions [0 1 2 3 4 5]
$ variables in same file [T] or not [F]
$
 3 4
 0 1 2
 T
$
$ -------------------------------------------------------------------- $
$ File prefix
$ number of characters in date [0(nodate),4(yearly),6(monthly),8(daily),10(hourly)]
$ IX and IY ranges [regular:IX NX IY NY, unstructured:IP NP 1 1]
$
 ww3.
 6
 1 1000000 1 1000000
$
$ For each field and time a new file is generated with the file name
$ ww3.date_xxx.nc , where date is a conventional time indicator with S3
$ characters, and xxx is a field identifier.
$
$ -------------------------------------------------------------------- $
$ End of input file                                                    $
$ -------------------------------------------------------------------- $


So I only changed 1 line in each file, the "HS LM," since I only want the significant wave height and wave length. (I have also tried "HS WLM"). In both cases I still get returned:

 Files for 1968/06/06 04:48:00 UTC: 

Writing new record hs       number     17 for 1968:06:06T04:48:00 in file ww3.196806.nc

Writing new record fp       number     17 for 1968:06:06T04:48:00 in file ww3.196806.nc

Writing new record dir      number     17 for 1968:06:06T04:48:00 in file ww3.196806.nc

And I am unable to see the mean wavelength as output. Can you give me any pointers as to what I am doing wrong, as it seems like it's not registering the changes at all, since it's also giving me frequency output? (I apologize as I know this must be a dumb question, I appreciate your help)

**Reply by japham0** [reply]
** I was able to output the correct changes by editing the ww3_shel.nml and ww3_ounf.nml files. It seems that changing the .inp files didn't have any effect- can I please ask what is the difference in functionality between the ww3_shel.nml and ww3_ounf.inp files?

If I may also ask, from my understanding HS is computed as HS=4\sqrt{E}, where E is the spectral average defined on pg.31 of the manual, and LM = 2pi/k. 

After discretizing the larger domain into a grid, are the values of F(f_r, theta) which are required to solve for E stored along the gridpoints? Similarly, are the values for k stored along the gridpoints as well? I've looked through the files and switches required to run the example regtest, and I am not sure where these values are computed or stored. While I see some of the switches denote certain numerical schemes to be used, where is the regtest finding stored values required to carry out the schemes?

Thanks again for your time and help.

**Comment by mickaelaccensi** [comment]
since the is 2 ways to provide the configuration to the WW3 programs (.inp or .nml), if the ww3_shel.nml is present it will be used in priority instead of ww3_shel.inp. So, you should use the .nml configuration file instead. In the future, the .inp configuration file should be remove from WW3.

let me know it works for you.

**Reply by japham0** [reply]
Thank you, modifying ww3_shel.nml did the trick.

Can I follow up and ask, which of the files in the regtest stores the values of F(f_r, theta), which from my understanding are necessary to compute HS? And similarly, is there another file in the regtest which stores the wavenumbers k which are necessary to compute LM?
